sdk/keyvault/azkeys/client.go

```diff
@@ -760,21 +760,22 @@ func updateKeyPropertiesFromGenerated(g generated.KeyVaultClientUpdateKeyRespons
 
 // UpdateKeyProperties updates the management properties of a key, but not its cryptographic material.
 // Pass nil for options to accept default values.
-func (c *Client) UpdateKeyProperties(ctx context.Context, key Key, options *UpdateKeyPropertiesOptions) (UpdateKeyPropertiesResponse, error) {
-	name, version := "", ""
-	if key.Properties != nil && key.Properties.Name != nil {
-		name = *key.Properties.Name
-	}
-	if key.Properties != nil && key.Properties.Version != nil {
-		version = *key.Properties.Version
+func (c *Client) UpdateKeyProperties(ctx context.Context, properties Properties, options *UpdateKeyPropertiesOptions) (UpdateKeyPropertiesResponse, error) {
```

Comment:
How/where do they update keyOps: https://docs.microsoft.com/en-us/dotnet/api/azure.security.keyvault.keys.keyclient.updatekeyproperties?view=azure-dotnet#azure-security-keyvault-keys-keyclient-updatekeyproperties(azure-security-keyvault-keys-keyproperties-system-collections-generic-ienumerable((azure-security-keyvault-keys-keyoperation))-system-threading-cancellationtoken) ?

Reply:
Thanks for pointing this out! Looks like it was never possible, although it could be in the current API because the `Key` type carries the operations. Why did we choose to add another parameter for the operations instead of adding them to the properties object?

Reply:
The permissible operations can't be updated in all the same places properties can.

Reply:
Are you thinking of import, which also takes `Properties` but gets the operations from the JWK?

Reply:
Partly, but in C# - and some other languages I think as well - our _properties_ object is the same one that hangs off a `KeyVaultKey`. In hindsight, probably over-sharing the type.

Reply:
That's the design here as well. I see we have some additional over-sharing in Go: several options bags have `Properties` fields and therefore have redundant fields and read-only options. I'll flatten these in another PR (tracked by #18324).